### PR TITLE
Chore: Cleanup Text and TextMetrics

### PR DESCRIPTION
**Pull request by SuperSodaSea**
##### Description of change

- `Text.graphemeSegmenter` is moved to `TextMetrics.graphemeSegmenter`, so that we can use it in `TextMetrics.wordWrapSplit`
    - Fortunately there is no new release since #8893, so no breaking change :)
- `Text.experimentalLetterSpacing` is moved to `TextMetrics.experimentalLetterSpacing`
- Add `IAdapter.getCanvasRenderingContext2D`
- Fix `TextMetrics.wordWrap`, now it support characters with multiple code points
    - Before: https://pixiplayground.com/#/edit/xFjAqYprZeC30XfEqbK_5
    - After: https://pixiplayground.com/#/edit/MPDAY7vYU-o3zgYn_FOf2
    - After (with `TextMetrics.experimentalLetterSpacing = true`): https://pixiplayground.com/#/edit/THqQ6nbHZmXDyqSnZBoEk
    - See also https://github.com/pixijs/pixijs/issues/5571#issuecomment-1328875757
- Move static initialization that is outside of `TextMetrics` into its class body
- Some small fixes in comment

##### Pre-Merge Checklist

- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

**Review by bigtimebuddy — APPROVED** [review]
Good adjustments. I agree with the organization here.

**Comment by SuperSodaSea** [comment]
Seems that there is a strange wrap between `🙆🏻‍♂️` and `😀`, need some investigation...

File: packages/text/src/TextMetrics.ts [hunk 1]
```diff
@@ -58,25 +97,77 @@ export class TextMetrics
...
+            const segmenter = new (Intl as IIntl).Segmenter();
+
+            return (s: string) => [...segmenter.segment(s)].map((x) => x.segment);
+        }
+
+        return (s: string) => [...s];
+    })();
+
+    /** Cache of {@see PIXI.TextMetrics.FontMetrics} objects. */
+    private static _fonts: { [font: string]: IFontMetrics } = {};
```
Comment (bigtimebuddy) [inline packages/text/src/TextMetrics.ts:142]: Could you change this type to `Record<string, IFontMetrics>`?

**Comment by bigtimebuddy** [comment]
@SuperSodaSea Is there more you want to do to this?

**Comment by SuperSodaSea** [comment]
@bigtimebuddy Seems that when measuring character with multiple code points using non-zero `letterSpacing`, there will be some extra `letterSpacing` being calculated. I need to fix this first.

https://github.com/pixijs/pixijs/blob/39b2ae6f78394d8211ad59e0b0024ce6667191fa/packages/text/src/TextMetrics.ts#L234

**Review by bigtimebuddy — APPROVED** [review]
Nice work!

**Pull request merged**